src/tezos/TezosOperations.ts

```diff
@@ -169,6 +170,32 @@ export namespace TezosOperations {
         }
     }
 
+    export async function appendRevealOperation(
+        network: string,
+        keyStore: KeyStore,
+        fee: number,
+        account: TezosTypes.Account,
+        operations: Operation[]
+    ) {
+        const isManagerKeyRevealed = await isManagerKeyRevealedForAccount(network, keyStore)
+        var returnedOperations: Operation[] = operations;
+        if (!isManagerKeyRevealed) {
+            const revealOp: Operation = {
+                kind: "reveal",
+                source: keyStore.publicKeyHash,
+                fee: fee.toString(),
+                counter: (Number(account.counter) + 1).toString(),
+                gas_limit: '120',
+                storage_limit: '0',
+                public_key: keyStore.publicKey
+            };
+            var operation = operations[0]
+            operation.counter = (Number(account.counter) + 2).toString()
+            returnedOperations = [revealOp, operation]
```

Comment:
Can we just do a regular array append?

Reply:
Initially I wanted to do that, but the issue is either we abstract the definition of originations/delegations/transactions to the helper function, making the function an unwieldy mess, or we keep them in the original functions. The problem is the helper function is where the reveal is checked to have happened or not, so you can't just append to the original operations list. You have to edit the inputted operation's counter, to be consistent